geosys/processing/geosys_processing_algorithm.py

```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
+
+"""
+/***************************************************************************
+ GeosysProcessingProvider
+                                 A QGIS plugin
+ GeosysProcessingProvider
+ Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                              -------------------
+        begin                : 2019-03-30
+        copyright            : (C) 2019 by Kartoza Pty. Ltd
+        email                : rohmat@kartoza.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from PyQt5.QtCore import QCoreApplication, QDate
+from PyQt5.QtWidgets import QDateEdit
+from processing.gui.wrappers import WidgetWrapper
+from qgis.core import (QgsProcessing,
+                       QgsFeatureSink,
+                       QgsProcessingAlgorithm,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingParameterFolderDestination,
+                       QgsProcessingParameterEnum,
+                       QgsProcessingParameterString)
+
+from geosys.bridge_api.definitions import INSEASON_MAP_PRODUCTS, SENSORS
+
+__copyright__ = "Copyright 2019, Kartoza"
+__license__ = "GPL version 3"
+__email__ = "rohmat@kartoza.com"
+__revision__ = "$Format:%H$"
+
+
+class DateWidgetWrapper(WidgetWrapper):
+    """WidgetWrapper for QgsProcessingParameterString that create and manage
+    a QDateEdit widget.
+    """
+    def createWidget(self):
+        """Override method."""
+        self.date_edit = QDateEdit()
+        self.date_edit.setCalendarPopup(True)
+        return self.date_edit
+
+    def setValue(self, value):
+        """Override method."""
+        date = QDate.fromString(value, 'yyyy-MM-dd')
+        self.date_edit.setDate(date)
+
+    def value(self):
+        """Override method."""
+        return self.date_edit.date().toString('yyyy-MM-dd')
+
+
+class MapCoverageDownloader(QgsProcessingAlgorithm):
+    """
```

Comment:
First line of docstring should fit in a single line followed by two line breaks then extended description.